wapitiCore/attack/mod_check_http_headers.py

```diff
@@ -0,0 +1,88 @@
+import requests
+from wapitiCore.attack.attack import Attack
+from wapitiCore.net.web import Request
+from wapitiCore.language.vulnerability import Additional
+
+class mod_check_http_headers(Attack):
+
+    name = "check_http_headers"
+
+    def check_xframe_option(self, request: object):
+        check_list = ['DENY', 'SAMEORIGIN', 'ALLOW-FROM']
+        if 'X-Frame-Options' not in request.headers:
+            return False
+        elif any(element in request.headers['X-Frame-Options'] for element in check_list):
+            return True
+        else:
+            return False
+
+    def check_xss_protection(self, request: object):
+        if 'X-XSS-Protection' not in request.headers:
+            return False
+        elif '1' in request.headers['X-XSS-Protection']:
+            return True
+        else:
+            return False
+
+    def check_content_type_options(self, request: object):
+        if 'X-Content-Type-Options' not in request.headers:
+            return False
+        elif 'nosniff' in request.headers['X-Content-Type-Options']:
+            return True
+        else:
+            return False
+
+    def check_hsts(self, request: object):
+        if 'Strict-Transport-Security' not in request.headers:
+            return False
+        elif 'max-age=' in request.headers['Strict-Transport-Security']:
+            return True
+        else:
+            return False
+
+    def attack(self):
+        url = self.persister.get_root_url()
+        session = requests.session()
+        req = session.get(url)
+        request = Request(url)
+        self.log_blue('Checking X-Frame-Options :')
+        if not self.check_xframe_option(req):
+            self.log_red('X-Frame-Options is not set')
+            self.add_addition(
+                category=Additional.MSG_HTTP_HEADERS,
+                level=Additional.LOW_LEVEL,
+                request=request,
+                info="X-Frame-Options is not set",
```

Comment:
You should use _("...") notation to support translations

Reply:
Generate error